app/src/androidTest/java/com/github/sdp_begreen/begreen/firebase/FirebaseAuthTestNoUILinked.kt

```diff
@@ -0,0 +1,86 @@
+package com.github.sdp_begreen.begreen.firebase
+
+import androidx.test.espresso.matcher.ViewMatchers.assertThat
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.MediumTest
+import com.github.sdp_begreen.begreen.rules.CoroutineTestRule
+import com.google.firebase.auth.ktx.auth
+import com.google.firebase.database.ktx.database
+import com.google.firebase.ktx.Firebase
+import com.google.firebase.storage.ktx.storage
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.drop
+import kotlinx.coroutines.flow.take
+import kotlinx.coroutines.flow.toList
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.tasks.await
+import kotlinx.coroutines.test.runTest
+import org.hamcrest.CoreMatchers.*
+import org.hamcrest.Matchers.contains
+import org.junit.BeforeClass
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+/**
+ * Testing [FirebaseAuth] needed to be split in half in order to be tested, due to problem
+ * with coroutine test environment and UI modification
+ *
+ * So this test class is responsible to test getting the user id which doesn't require
+ * to start any activities
+ */
+@OptIn(ExperimentalCoroutinesApi::class)
+@RunWith(AndroidJUnit4::class)
+@MediumTest
+class FirebaseAuthTestNoUILinked {
+    companion object {
+        private val firebaseAuth = FirebaseAuth()
+        @BeforeClass
+        @JvmStatic fun setup() {
+            try {
```

Comment:
Code duplication, you can put the Firebase emulation setup in a central testing utility file

Reply:
That is true, thanks for pointing it out, we will change that.